MonkeyWrench.Web.WebService/Authentication.cs

```diff
@@ -127,13 +127,24 @@ private static void VerifyAnonymousAllowed()
 
 		public static void VerifyUserInRole (HttpContext Context, DB db, WebServiceLogin login, string role, bool @readonly)
 		{
+			VerifyUserInRoles(Context, db, login, new string[] { role }, @readonly);
+		}
+
+		public static void VerifyUserInRoles (HttpContext Context, DB db, WebServiceLogin login, string[] roles, bool @readonly)
+		{
 			WebServiceResponse dummy = new WebServiceResponse ();
 			Authenticate (Context, db, login, dummy, @readonly);
 
-			if (!dummy.IsInRole (role)) {
-				log.InfoFormat ("The user '{0}' has the roles '{1}', and requested role is: {2}", login.User, dummy.UserRoles == null ? "<null>" : string.Join (",", dummy.UserRoles), role);
-				throw new UnauthorizedException ("You don't have the required permissions.");
+			foreach (var role in roles) {
```

Comment:
Good point. If it's blank it should be okay, but if someone mistypes with an extra comma there could be problems. There should at least be a empty/null check for that case. I'll add it.